realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -45,7 +45,7 @@
  * @see io.realm.Realm
  * @see io.realm.DynamicRealm
  */
-abstract class BaseRealm implements Closeable {
+public abstract class BaseRealm implements Closeable {
```

Comment:
If you have any ideas for improving it I am all ears though. Exposing the configuration so adapters can open their own instance is not really possible either as there is no lifecycle callback on an Adapter when the view is no longer attached :(